fe/fe-core/src/main/java/com/starrocks/mysql/privilege/DbPrivTable.java

```diff
@@ -41,14 +43,15 @@ public class DbPrivTable extends PrivTable {
      * saved in 'savedPrivs'.
      */
     public void getPrivs(UserIdentity currentUser, String db, PrivBitSet savedPrivs) {
+        List<PrivEntry> userPrivEntryList = map.get(currentUser);
```

Comment:
Should all the getPrivs() methods of UserPriveTable/TablePriveTable/DbPrivTable/ResourcePrivTable use getReadOnlyIteratorByUser to get the the entiryLIst?

Reply:
This is a legacy bug. I intended to fix in another PR. Right now the logic remains as before.